docs/storage-design.md

```diff
@@ -0,0 +1,149 @@
+# Storage and Document Management in Alkemio
+
+This document is created to capture the design requirements and decisions made for Document Management in Alkemio.
+
+
+## Requirements
+The following are high priority desired functionalities related to document management on Alkemio:
+* **Secure Access**: Allow uploading and having links to documents that are protected by Alkemio's authorization framework
+* **Management**: Allow admins to see documents that have been uploaded in their hub, as well as meta information about thos documents
+* **Secure**: Be able to track who uploaded a file and when, plus potentially additional meta information (e.g. upload IP)
+* **Uploading rights per context**: Allow users to upload documents where they have relevant authorization privileges
+* **Accessible via a URL**: the files need to be requestable via a http request whereby the response is the actual document
+* **Quotas**: there should be limits on the amount of storage space used within a given context. This can then later be controlled by license rights. 
+* **Single protocol**: it should be possible to disable the external ipfs protocol usage
+
+The following non-functional requirements apply:
+* **Performant**: The files access should be as fast as feasible given we are serving the content from our own servers. 
+  
+The following security requirements should be designed for:
+* **Encryption at rest**: allowing the files to be encrypted at Rest
```

Comment:
This is very vague, also REST is the API layer, encryption is the service layer. They are not coupled so I don't think this is relevant comment.

Reply:
Encryption at rest is a standard industry term as far as I know, but for clarity I have removed this phrase from this document largely.